cfgov/legacy/housing_counselor/generator.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import absolute_import, unicode_literals
+
+import glob
+import json
+import os
+import re
+import sqlite3
+
+from django.template import loader
+from math import acos, cos, radians, sin
+from six import print_
+
+
+def distance_in_miles(lat1_radians, lng1_radians, lat2_radians, lng2_radians):
+    """Estimate distance in miles between two points in radians.
+
+    Uses simplified algorithm to determine proximity based on latitude and
+    longitude, described at https://stackoverflow.com/q/1916953.
+    """
+    earth_radius_in_miles = 3959
+
+    return earth_radius_in_miles * acos(
+        cos(lat1_radians) *
+        cos(lat2_radians) *
+        cos(lng2_radians - lng1_radians) +
+        sin(lat1_radians) *
+        sin(lat2_radians)
+    )
+
+
+def get_db_connection():
+    connection = sqlite3.connect(':memory:')
+    connection.create_function('distance_in_miles', 4, distance_in_miles)
+
+    return connection
+
+
+def fill_db(connection, counselors):
+    def prepare_counselors(counselors):
```

Comment:
So it's not a ton: but we need to handle counselors without coordinates, which means geocoding them.